src/ruby/puppet-server-lib/puppet/server/auth_provider.rb

```diff
@@ -0,0 +1,13 @@
+## This is a "dummy" AuthProvider that Puppet Server registers with core
+## Puppet for cases where Puppet Server decides that it should be in charge of
+## authorizing requests at the Clojure / Ring handler level - depending upon
+## the configuration of the 'use_legacy_auth_conf' setting - and not core
+## Puppet.
+
+class Puppet::Server::AuthProvider
```

Comment:
Hm, we might not want this to be the same name as the Puppet class, although that may never technically bite us?